### PR TITLE
Spring 6 Upgrade!!!

### PR DESCRIPTION
**Pull request by msnwatson**
See below comment for PR description

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -51,26 +53,30 @@ tasks.named('jacocoTestReport').configure {
...
-
-  tasks.named('spectralLint').configure {
-    dependsOn 'forkedSpringBootStop'
-  }
+//  tasks.named('forkedSpringBootRun').configure {
+//    dependsOn 'bootStartScripts'
+//    dependsOn 'checkstyleIntegrationTest'
+//    dependsOn 'checkstyleMain'
+//    dependsOn 'checkstyleTest'
+//    dependsOn 'compileIntegrationTestJava'
```
Comment (msnwatson) [inline app/build.gradle:64]: commenting this out is probably not necessary once the upgrade is complete

File: app/build.gradle [hunk 2]
```diff
@@ -51,26 +53,30 @@ tasks.named('jacocoTestReport').configure {
 afterEvaluate {
   // Disable https://github.com/springdoc/springdoc-openapi-gradle-plugin
   // Not particularly useful
-  generateOpenApiDocs.enabled = false
-  forkedSpringBootRun.enabled = false
+//  generateOpenApiDocs.enabled = false
```
Comment (msnwatson) [inline app/build.gradle:56]: commenting this out is probably not necessary once the upgrade is complete

File: gradle-plugins/src/main/groovy/local.std.java.application-conventions.gradle [hunk 3]
```diff
@@ -19,7 +19,7 @@ plugins {
     // Use our `local` container variants
     id 'local.java.container-spring-conventions'
 
-    id 'starter.java.lint-spectral-conventions'
+//    id 'starter.java.lint-spectral-conventions'
```
Comment (msnwatson) [inline gradle-plugins/src/main/groovy/local.std.java.application-conventions.gradle:22]: this should get uncommented too

File: gradle.properties [hunk 4]
```diff
@@ -6,16 +6,14 @@ dockerRegistry=va
 # so disableSafeguard so that tasks from the included builds will be executed.
 taskinfo.disableSafeguard=true
 
-# 3.11.0 uses spring_boot 2.5.3
-# 3.12.0 uses spring_boot 2.5.5
-# 3.13 through 3.15 are not compatible with va.starter's spring_boot version
-camel_version=3.19.0
+camel_version=3.20.4
```
Comment (msnwatson) [inline gradle.properties:9]: obviously gotta update this

**Comment by msnwatson** [comment]
@agile-josiah can you post in the PR description the steps you've taken for testing the changes? Just cause this PR is so large and it will be easy to miss things.

**Comment by agile-josiah** [comment]
> @agile-josiah can you post in the PR description the steps you've taken for testing the changes? Just cause this PR is so large and it will be easy to miss things.

I shouldn't have taken this PR out of draft mode. I think there's still more work to do here. I will start by building locally and posting the steps I took to test as you suggested.

**Comment by agile-josiah** [comment]
## What was the problem?

An upgrade to Spring 6 required an update to dependencies and some library functions have been deprecated and needed to be changed. 
Associated tickets or Slack threads:
- This is an effort to increase capability of hoppy library for VRO partners in pursuit of this task:
https://github.com/department-of-veterans-affairs/abd-vro/issues/1802

## How does this fix it?[^1]
This PR will add the changes required to complete the upgrade to Camel 4, Spring Boot 3, and Spring 6.
- Most of the changes are related to javax -> jackson api changes and httpSecurityChain changes.
## How to test this PR
Pull the PR and install the hoppy library:
1. run ./gradlew clean
2. run ./gradlew check build docker
3. run ./gradlew dockerComposeUp
4. After the app is built and running access this url [in](http://localhost:8111/actuator/health) your browser. 



<details>
<summary>Optionally, check the additional actuator urls for further verification.</summary>

``` Health - /actuator/health
Shows application health information.

Info - /actuator/info
Displays arbitrary application info.

Metrics - /actuator/metrics
Shows 'metrics' information for the current application.

Loggers - /actuator/loggers
Shows and modifies the configuration of loggers in the application.

Heapdump - /actuator/heapdump
Returns a GZip compressed hprof heap dump file.

Threaddump - /actuator/threaddump
Performs a thread dump.

Env - /actuator/env
Exposes properties from Spring's ConfigurableEnvironment.

Beans - /actuator/beans
Displays a complete list of all the Spring beans in the application.

Mappings - /actuator/mappings
Displays a collated list of all @RequestMapping paths.

Scheduledtasks - /actuator/scheduledtasks
Displays the scheduled tasks in the application.
```
</details>

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

**Comment by agile-josiah** [comment]
it's finally over.

**Review by agile-josiah — APPROVED** [review]
Approving on behalf of @msnwatson. He is the author of the PR so he cannot review it.

**Comment by agile-josiah** [comment]
@chengjie8 I addressed all your suggested changes. We should be good to go!

**Review by chengjie8 — APPROVED** [review]
LGTM. @msnwatson and @agile-josiah great work!!!

**Pull request merged**